File: simple-stack-example-mvp/src/main/java/com/zhuinden/simplestackdemoexamplemvp/application/Injector.java
package com.zhuinden.simplestackdemoexamplemvp.application;

import com.zhuinden.simplestackdemoexamplemvp.application.injection.SingletonComponent;

/**
 * Created by Zhuinden on 2017.03.27..
 */

public class Injector {
    private Injector() {
    }

    public static SingletonComponent get() {
        return CustomApplication.get().getComponent();
    }
}

File: examples/simple-stack-example-mvp-view/src/main/java/com/zhuinden/simplestackdemoexamplemvp/application/Injector.kt
package com.zhuinden.simplestackdemoexamplemvp.application

import com.zhuinden.simplestackdemoexamplemvp.application.injection.SingletonComponent

/**
 * Created by Zhuinden on 2017.03.27..
 */

object Injector {
    @JvmStatic
    fun get(): SingletonComponent = CustomApplication.get().component()
}
